### DISCUSSION TITLE
Murex in Direnv

### DISCUSSION THREAD

**Discussion by tiymat** [lmorg/murex#795] [Show and tell]
Hey all, just wanted to mention that last week I was able to get my PR to add Murex support into [Direnv](https://github.com/direnv/direnv) merged, and it is present in the current Direnv release.

Direnv allows the creation and management of directory-specific environment variables, which I've found very useful in my work. I've added a brief example here to the creation of an environment variable `FOO` that only exists within a particular directory.
![image](https://github.com/lmorg/murex/assets/138939221/1c0f9341-6fcb-4d2d-b0b7-5a14299f6122)

Thanks to @lmorg for the fixes in v6.0 and before that made this possible!

**Comment by lmorg** [comment]
Oh that’s cool. Well done with the PR too

**Comment by lmorg** [comment]
Looking through the code, the interesting thing I notice is that, out of all of the shells, the Murex code is both the most terse yet also the most readable too (in my admittedly biased opinion) .

There's definitely still a lot of footguns in Murex that need ironing out but, and at risk of being self-congratulatory, I think it's a real credit to everyone who's contributed that we've achieved something which can be both succinct as scripting language while still easy to read. It's a hard balance to make.

**Reply by tiymat** [reply]
I guess that repo is a great Rosetta Stone for seeing different shell scripts doing the same thing. I didn't pay much attention to the others while I was implementing it, but looking at them now I'd have to agree; I think you can be very proud of Murex. Even though it's a relatively small code snippet it's a nice example because it shows off one of Murex's strengths, handling JSON/typed data.

I wrote the PR because
1. Direnv is a great tool I use frequently
2. I wanted to improve my familiarity/comfort with Murex
3. The test suite they use was also useful in finding a couple of bugs in Murex's variable handling
4. I think having the docs mention Murex as a supported shell might drive awareness or interest in Murex

Here's hoping that fourth point pays off :smile:

**Reply by lmorg** [reply]
I'm sure it will :)
